.github/workflows/contribs.yml

```diff
@@ -1,30 +1,54 @@
-name: contribs
+name: Contribs
 
 on:
   push:
     branches:
       - master
-  workflow_dispatch:
   pull_request:
+    paths:
+      - contribs/**
+  workflow_dispatch:
 
 jobs:
   setup:
     runs-on: ubuntu-latest
     outputs:
       programs: ${{ steps.set-matrix.outputs.programs }}
+      go-versions: ${{ steps.get-go-versions.outputs.go-versions }}
     steps:
       - uses: actions/checkout@v4
+
       - id: set-matrix
-        run: echo "::set-output name=programs::$(ls -d contribs/*/ | cut -d/ -f2 | jq -R -s -c 'split("\n")[:-1]')"
+        run: |
+          echo "::set-output name=programs::$(ls -d contribs/*/ | cut -d/ -f2 | jq -R -s -c 'split("\n")[:-1]')"
+
+      - id: get-go-versions
+        run: |
+          contribs_programs=$(ls -d contribs/*/ | cut -d/ -f2)
+          versions_map="{"
+          
+          for p in $contribs_programs; do
+            # Fetch the go version of the contribs entry, and save it
+            # to a versions map we can reference later in the workflow
+            go_version=$(grep "^go [0-9]" contribs/$p/go.mod | cut -d ' ' -f2)
+            versions_map="$versions_map\"$p\":\"$go_version\","
+          done
+          
+          # Close out the JSON
+          versions_map="${versions_map%,}"
+          versions_map="$versions_map}"
+          echo "::set-output name=go-versions::$versions_map"
+
```

Comment:
shall we force to all contribs to use the same standardized go version to avoid problems and keep it simple? IMO if we have things living in the same monorepo they supposedly are living together to keep them on sync somehow.

Reply:
Honestly, I don't really get the need to not use the latest version of go on the monorepo

We can open up this pandora's box on a call sometime, for now I'd like to keep the go version meddling outside this PR 🙏